lib/git/lib.rb

```diff
@@ -310,7 +311,7 @@ def branches_all
     def list_files(ref_dir)
       dir = File.join(@git_dir, 'refs', ref_dir)
       files = []
-      Dir.chdir(dir) { files = Dir.glob('**/*').select { |f| File.file?(f) } } rescue nil
+      files = Dir.glob(File.join(dir, '**/*')).select { |f| File.file?(f) }
```

Comment:
I wonder if the `rescue nil` removal can be dangerous? Unfortunately not aware of its root cause of being there though.